vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,400 @@
+a given area) or not (i.e. project data representing plot points in a park).
+
+If the former, the coordinates need to be in order. If the later,
+
+```{r, createlocs, results = 'hide'}
+  # points on a regular spaced grid in the WGS84 CRS
+  locationsGrid <- data.frame(X_WGS84 = c(rep(-124.5938, 5), rep(-124.5312, 5)),
+                        Y_WGS84 = rep(c(47.90625, 47.96875, 48.03125, 48.09375,
+                                     48.15625), 2))
+
+  # irregularly spaced points in the WGS84 CRS
+  #locationsRandom
+```
+
+## Dimension Data
+
+NetCDFs are organized into dimensions. Dimensions have two key components: Bounds
+and values. The dimensions also have other information which is degined in the Attibute Data
+(see below).
+
+### Latitude (Y) and Longitude (X) Dimensions
+
+All NetCDFs will have at least two dimensions, latitude and longitude.
+
+The bounds and values of the latitude and longitude are determined from the location
+data within the *create_empty_netCDF_file* function.
+
+### Time (T) Dimension
+
+The user must define the bounds and values for the time dimension, as inputs to the
+*create_empty_netCDF_file*.
+
+The bounds of the time dimension are the start and end of each interval of the time dimension.
+
+The values refer to the midpoint of each interval of the time dimension.
+
+The bounds and values need to be in the same units.
+
+A common unit for netCDF is days since 1900. A full list of accepted time units can
+be found [here](https://ncics.org/portfolio/other-resources/udunits2/).
+
+```{r, timeD}
+
+    ## time vals
+    time_midpoints <- c(43920, 44285, 44650, 45015, 45381, 45746, 46111,
+                          46476, 46842, 47207)# mid point of year n days since 1900-01-01
+
+    ## time bounds
+    annual_time_bounds <- c(43737, 44102, 44102, 44467, 44467, 44832, 44833, 45197,
+                   45197, 45563, 45563, 45928, 45928, 46293, 46293, 46658,
+                  46658, 47024, 47024, 47389) # beginning and end of year in days since 1900-01-01
+```
+
+### Depth (Z) Dimension
+
+The user must define the bounds and values for the time dimension.
+
+```{r, depthD}
+
+    ## depth vals
+    depths_max <- c(5, 10, 15) # maximum depth of each layer of depth
+
+    ## depth bounds
+    depth_bounds <- c(0, 5, 5, 10, 10, 15) # min and max of each nlayer of depth depth
+```
+
+## Attribute Data
+
+Attribute data pertains to the metadata contained in the netCDFs. NetCDFs are
+self-documenting and require metadata attributes, providing information
+necessary for interpreting the data.
+
+Different attribute types include variable attributes, time attributes, depth
+attributes, and global attributes. The format of attributes is a named list. Each attribute list
+has a bare minimum of required fields. Endless other fields are optional and can
+be added by the user as they see fit.
+
+### Variable Attributes
+
+A named list describing the variable data. The *name* and *units* fields are required.
```

Comment:
As for the function documentation, I think we should encourage using CMIP6 standard names where available and link to variable lists, e.g., the  [day](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_day.json) and [Eday](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Eday.json) tables for time-varying variables and to [fx](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_fx.json) and [Efx](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Efx.json) for time-invariant variables.

Reply:
fixed